bpf/bpf_lxc.c

```diff
@@ -161,9 +174,15 @@ static inline int ipv6_l3_from_lxc(struct __sk_buff *skb,
 		return DROP_POLICY;
 	}
 
-	if (state) {
-		ret = lb_dsr_dnat(skb, state, tuple);
-		cilium_trace(skb, DBG_GENERIC, state, ret);
+	/* Service redirection must occur before conntrack to ensure that the
+	 * conntrack entry is created for the translated final endpoint. */
+	if (!skip_service && (svc = lb6_lookup_service(skb, &key)) != NULL) {
```

Comment:
Good catch, it was before conntrack before so the comment became outdated. I think it makes sense to eventually do it before conntrack which would allow to see the actual backend service in the CT table at the source. The reverse translation becomes slightly more complex though.